wandb/sdk/internal/internal_api.py

```diff
@@ -1887,11 +1887,20 @@ def upload_file(
             response_content = e.response.content if e.response is not None else ""
             logger.error(f"upload_file response body: {response_content}")
             status_code = e.response.status_code if e.response is not None else 0
+            # S3 reports retryable request timeouts out-of-band
+            is_aws_retryable = (
+                status_code == 400 and "RequestTimeout" in response_content
+            )
```

Comment:
This feels dangerous, let me see if we can find a more reliable way to detect this...

Reply:
as discussed elsewhere, let's add checking `x-amz-meta-md5` and get this one merged.

Reply:
done